ui/k-toolbar/main.kytos

```diff
@@ -41,6 +54,161 @@
+                  _this.interface_names = _interface_names;
+              });
+
+              request.fail(function( jqXHR, textStatus ) {
+                  alert("Request failed: " + textStatus);
+              });
+            },  
+            /*
+                Gets the Link information using the Get api call from the topology api.
+            */
+            loadLinks: function(){
+                let _link_names = [];
+                var _this = this;
+                var request = $.ajax({
+                    url: this.$kytos_server_api + "kytos/topology/v3/links",
+                    type: "GET",
+                    dataType: "json",
+                    contentType: "application/json"
+                });
+                request.done(function(data) {
+                    let links = data.links;
+                    $.each(links, function(i, li) {
+                        if(li.metadata.link_name){
+                            _link_names.push({"value": li.id, "description": li.metadata.link_name})
+                        }
+                        else{
+                            _link_names.push({"value": li.id, "description": li.id})
+                        }
+                    });
+                    _this.link_names = _link_names;
+                });
+               request.fail(function( jqXHR, textStatus) {
+                   alert("Request failed: " + textStatus)
+               });
+            },
+            /*
+                This method takes the various lists with switch, link, and interface information 
+                and combines the options the user selected to JSON format.
+            */
+            formatInfo: function(){
+                var lengthOne = this.link_items.length;
+                for(let i = 0; i < lengthOne; i++){
+                    var item = this.link_items[i]
+                    let links = {
+                        "id": item
+                    }
+                    this.merged_list.push(links)
+                }
+
+                var lengthTwo = this.interface_items.length;
+                for(let i = 0; i < lengthTwo; i++){
+                    var itemTwo = this.interface_items[i]
+                    let interfaces = {
+                        "interface_id": itemTwo
+                    }
+                   this.merged_list.push(interfaces)
+                }
+                this.merged_list = this.merged_list.concat(this.items)
+           }
+        },       
+        data() {
+            return{
+                id: "",
+                description: "",
+                start: "",
+                end: "",
+                items: [],
+                interface_items: [],
+                status: "",
+                dpid_names: [],
+                interface_names: [],
+                link_names: [],
+                link_items: [],
+                merged_list: [],
+           }
+       },
+       mounted(){
+           this.loadNames();
```

Comment:
Since you're using both `/switches` and `links/` endpoint you can fire up these two functions asynchronously with promises, check out [this example that Aris has used recently](https://github.com/Aduan002/maintenance/blob/edit-and-delete-maintenance-window/ui/k-info-panel/edit_window.kytos#L384-L385), could fetch both of them asynchronously?